src/libraries/System.Private.Xml/src/System/Xml/XPath/Internal/QueryBuilder.cs

```diff
@@ -462,7 +466,7 @@ private Query ProcessNode(AstNode root, Flags flags, out Props props)
             }
 
             Debug.Assert(root != null, "root != null");
-            Query result = null;
+            Query? result = null;
```

Comment:
I think if you remove this assignment this can be marked as non-null

Reply:
(or assign null! if Debug.Fail is not recognized by compiler as not-returning)

Reply:
Yes, null assignment is not needed.

Reply:
Actually, this can be null for `case AstNode.AstType.Operator`; method `ProcessOperator` returns null when `root.OperatorType == Operator.Op.INVALID`.